shared/src/Datadog.Trace.ClrProfiler.Native/log.h

```diff
@@ -23,6 +23,34 @@ class Log final
         inline static const shared::WSTRING folder_path = WStr(R"(Datadog .NET Tracer\logs)");
 #endif
         inline static const std::string pattern = "[%Y-%m-%d %H:%M:%S.%e | %l | PId: %P | TId: %t] %v";
+
+        static bool enable_buffering() {
+#ifdef _WIN32
+            // For Windows SSI, we are injected into all .NET Core processes, so we can't enable flushing of logs
+            // until we know that the process is going to be instrumented, otherwise we will create way
+            // too many log files. Note that we don't currently distinguish between .NET FX and .NET Core, because
+            // we need to make the decision _before_ we have that information.
+            const auto isSingleStepVariable = shared::GetEnvironmentValue(EnvironmentVariables::SingleStepInstrumentationEnabled);
+            if (isSingleStepVariable.empty())
+            {
+                // not in Windows SSI, so we don't need buffering
+                return false;
+            }
+
+            const auto bufferEnvVar = shared::GetEnvironmentValue(EnvironmentVariables::LogBufferingEnabled);
+
+            // enable buffering _unless_ the variable is present and set to false
+            // if the variable is not set, or it's set to false
+            bool enable_buffering;
+            return bufferEnvVar.empty()
+                || !shared::TryParseBooleanEnvironmentValue(bufferEnvVar, enable_buffering)
+                || enable_buffering == true;
```

Comment:
Just to be sure I understand. We will enable buffering if
- `EnvironmentVariables::LogBufferingEnabled` is not present (default behavior)
- `EnvironmentVariables::LogBufferingEnabled` is set but not parseable
- `EnvironmentVariables::LogBufferingEnabled` is set to true

Reply:
Correct!

Put another way, the _only_ way will _not_ enable buffering in SSI on Windows is if `EnvironmentVariables::LogBufferingEnabled` is set to `false` (or equivalent)